### ISSUE TITLE
Check for required information prior to release step

### ISSUE THREAD

**Issue by horgh** [ghedo/p5-Dist-Zilla-Plugin-GitHub#50]
I use these plugins as part of a bundle, and I found at least `GitHub::Update` checks for some information it needs only after the release, and then errors if it does not find it. It would be good to check up front and not let the release occur if it will cause an error. That way we could fix these problems before we have many side effects.

Here's a snippet of what I saw:

```
[@MAXMIND/UploadToCPAN] POSTing upload for WebService-PayPal-PaymentsAdvanced-0.000023.tar.gz to https://pause.perl.org/pause/authenquery                                             
[@MAXMIND/UploadToCPAN] PAUSE add message sent ok [200]                                    
[@MAXMIND/GitHub::Update] Err: Missing value 'github.user' in git config                   

*** Please tell me who you are.              

Run                                          

  git config --global user.email "you@example.com"                                         
  git config --global user.name "Your Name"  

to set your account's default identity.      
Omit --global to set the identity only in this repository.                                 

fatal: unable to auto-detect email address (got 'root@12f515da5e27.(none)')
```

Thank you!

**Comment by horgh** [comment]
I made a PR about this. However after looking at the code I believe the critical problem I showed in the snippet above was not due to this plugin.

My release failed because of missing Git email/name, whereas the GitHub::Update complaint is about the GitHub user. I still think it would be good to require GitHub user be present though, so I made a PR here anyway.

I'll have to track down what plugin was trying to use Git and lead to the fatal error.

**Comment by karenetheridge** [comment]
> I'll have to track down what plugin was trying to use Git and lead to the fatal error.

Yes please - this is would be the right plugin to do a check for a valid user config.

**Comment by afresh1** [comment]
I believe that the GitHub::Create Plugin needs a BeforeMint check and GitHub::Update needs a BeforeRelease check that the authentication to GitHub works. Similar to what the [Git::Push Plugin does](https://github.com/karenetheridge/Dist-Zilla-Plugin-Git/blob/master/lib/Dist/Zilla/Plugin/Git/Push.pm#L54) to verify that you can connect before the action happens and those steps fail.
